test/fork/migration-v2.js

```diff
@@ -689,143 +683,309 @@ describe('V2 upgrade', function () {
+      for (let j = 0; j < idsOfProductsStakedInPS.length; j++) {
+        const productId = idsOfProductsStakedInPS[j];
+        const productAddress = addressesOfProductsStakedInPS[j];
 
-      for (const i of contractIds) {
-        if (i === -1) {
-          // contract was not migrated to v2 (deprecated and sunset)
+        // Product is deprecated and sunset (we didn't migrate it)
+        if (productId === -1) {
+          deprecatedProducts.add(productAddress);
           continue;
         }
-        const productPrice = await pooledStaking.getV1PriceForProduct(i);
+
+        const productPrice = await this.pooledStaking.getV1PriceForProduct(productId);
+
+        // Product is deprecated and not sunset (we migrated it as covers can still be claimed, but the quote engine
+        // can't give us a price for it)
         if (productPrice.toString() === MaxUint96) {
-          // it's not a supported product
+          deprecatedProducts.add(productAddress);
           continue;
         }
 
-        const stakedProduct = await stakingPool.products(i);
-
-        console.log(`Checking product with id: ${i}`);
+        // Product has no stake in PS
+        const stakeForProductInPS = await this.pooledStaking.stakerStoredContractStake(stakerAddress, productAddress);
+        if (stakeForProductInPS.isZero()) {
+          productsWithNoStake.add(productAddress);
+          continue;
+        }
 
-        console.log({
-          productPrice: productPrice.toString(),
-          stakedProductTargetPrice: stakedProduct.targetPrice.toString(),
-          stakedProductBumpedPrice: stakedProduct.bumpedPrice.toString(),
-        });
+        const stakedProduct = await stakingPool.products(productId);
         expect(stakedProduct.targetPrice).to.be.equal(productPrice.div(BigNumber.from((1e16).toString())));
         expect(stakedProduct.bumpedPrice).to.be.equal(productPrice.div(BigNumber.from((1e16).toString())));
+
+        // The expected weight is a number between 0-100
+        // equal to the ratio between the stake amount for the product in LegacyPooledStaking
+        // and the total deposit of that staker.
+        const expectedWeight = stakeForProductInPS
+          .mul(parseEther('1'))
+          .div(depositInPS[stakerAddress])
+          .div(parseEther('0.01'));
+        const expectedWeightCapped = expectedWeight.gt(MAX_STAKINGPOOL_PRODUCT_WEIGHT)
+          ? MAX_STAKINGPOOL_PRODUCT_WEIGHT
+          : expectedWeight;
+        expect(stakedProduct.targetWeight).to.be.equal(expectedWeightCapped);
+        expect(stakedProduct.lastEffectiveWeight).to.be.equal(BigNumber.from(0));
+      }
+    }
+
+    // Only managers of pools with non-zero deposits own a StakingNFT
+    const poolsWithDepositsCount = Object.values(depositsInStakingPools).filter(deposit => deposit.gt(0)).length;
+
+    console.log(`heck that the right number of StakingNFTs are minted: ${poolsWithDepositsCount}`);
```

Comment:
This number is 3 because the NexusMutual foundation is not migrated yet and we have

Hugh's Pool
Armor AAA Pool
Armor AA Pool

Code is written however so that when/if it gets a deposit, it will be incremented automatically (Deposits > 0.